src/core/providers/ogr/qgsogrconnpool.h

```diff
@@ -92,13 +50,31 @@ class QgsOgrConnPoolGroup : public QObject, public QgsConnectionPoolGroup<QgsOgr
     //! QgsOgrConnPoolGroup cannot be copied
     QgsOgrConnPoolGroup &operator=( const QgsOgrConnPoolGroup &other ) = delete;
 
+    ~QgsOgrConnPoolGroup() override
+    {
+      for ( const Item &item : std::as_const( conns ) )
+      {
+        destroyOgrConn( item.c );
+      }
+    }
+
+    void connectionCreate( const QString &connInfo, QgsOgrConn *&c ) override;
+    void connectionDestroy( QgsOgrConn *c ) override;
```

Comment:
```suggestion
    void connectionDestroy( QgsOgrConn *connection ) override;
```
... and other methods too